cmd/nomos/version/version.go

```diff
@@ -136,44 +140,61 @@ func filterConfigs(contexts []string, all map[string]*rest.Config) map[string]*r
 	return cfgs
 }
 
-func lookupVersionAndMode(ctx context.Context, cfg *rest.Config) (string, *bool, error) {
+func getImageVersion(deployment *v1.Deployment) (string, error) {
+	reconcilerImage := strings.Split(deployment.Spec.Template.Spec.Containers[0].Image, ":")
+	if len(reconcilerImage) <= 1 {
+		return "", fmt.Errorf("Failed to get valid image version from: %s", reconcilerImage)
+	}
+	return reconcilerImage[1], nil
+}
+
+func lookupVersionAndMode(ctx context.Context, cfg *rest.Config) (string, *bool, string, error) {
```

Comment:
Now the function returns multiple values. I would suggest adding a comment to explain what are those values.

Reply:
added explanation in function description